### PR TITLE
Lua themes with no options wouldn't load anymore (got broken in b4939…

### PR DESCRIPTION
**Pull request by kilrah**
Please check if that's not dangerous, but don't believe so...

### CODE REVIEW

**Comment by bsongis** [comment]
This was done this way in case there is no RAM left for the malloc (options is 0)

**Comment by kilrah** [comment]
Better now? Thing is options is also 0 when there are no options and we thus ask malloc to alloc 0 bytes...

File: radio/src/lua/widgets.cpp [hunk 1]
```diff
@@ -235,11 +235,13 @@ void luaLoadThemeCallback()
 
   if (name) {
     ZoneOption * options = createOptionsArray(themeOptions, MAX_THEME_OPTIONS);
-    LuaTheme * theme = new LuaTheme(name, options);
-    theme->loadFunction = loadFunction;
-    theme->drawBackgroundFunction = drawBackgroundFunction;
-    theme->drawTopbarBackgroundFunction = drawTopbarBackgroundFunction;
-    TRACE("Loaded Lua theme %s", name);
+    if (options || (themeOptions == 0)) {
```
Comment (bsongis) [inline radio/src/lua/widgets.cpp]: perhaps we could avoid to call `createOptionsArray` when themeOptions is NULL, no?

**Comment by bsongis** [comment]
I think this version would work:

```
  if (name) {
    ZoneOption * options = NULL;
    if (themeOptions) {
      options = createOptionsArray(themeOptions, MAX_THEME_OPTIONS);
      if (!options)
        return;
    }
    LuaTheme * theme = new LuaTheme(name, options);
    theme->loadFunction = loadFunction;
    theme->drawBackgroundFunction = drawBackgroundFunction;
    theme->drawTopbarBackgroundFunction = drawTopbarBackgroundFunction;
    TRACE("Loaded Lua theme %s", name);
  }
```

**Pull request merged**